src/libraries/System.Runtime.InteropServices/tests/LibraryImportGenerator.UnitTests/CompileFails.cs

```diff
@@ -209,5 +209,18 @@ public async Task ValidateDisableRuntimeMarshallingForBlittabilityCheckFromAssem
 
             TestUtils.AssertPostSourceGeneratorCompilation(newComp);
         }
+
+        [Fact]
+        public async Task ValidateRequireAllowUnsafeBlocksDiagnostic()
+        {
+            string source = CodeSnippets.TrivialClassDeclarations;
+            Compilation comp = await TestUtils.CreateCompilation(new[] { source }, allowUnsafe: false);
+            TestUtils.AssertPreSourceGeneratorCompilation(comp);
+
+            var newComp = TestUtils.RunGenerators(comp, out var generatorDiags, new Microsoft.Interop.LibraryImportGenerator());
+
+            // The errors should indicate the AllowUnsafeBlocks is required.
+            Assert.True(generatorDiags.All(d => d.Id == "SYSLIB1062"));
```

Comment:
Are we potentially issuing the diagnostic multiple times?

Reply:
We are issuing it multiple times - one for each compilation. Would you prefer to have only one?

Reply:
Does "each compilation" mean you'd end up getting one per LibraryImportAttribute usage?  If so, that could be a lot; dedup'ing down to a single one would be a better user experience IMO.

Reply:
Yes, this diagnostic will be once per project.

Reply:
Cool. Can the test then assert it's getting the expected count?